### PR TITLE
Test !nick when the user already has the nick

### PR DESCRIPTION
**Pull request by jaller94**
In preparation of #1018 I want the CI to ring an alarm, if changing the nick to your current nick causes a different notice to appear in Matrix.

### CODE REVIEW

File: spec/integ/admin-rooms.spec.js [hunk 1]
```diff
@@ -308,6 +309,77 @@ describe("Admin rooms", function() {
...
+
+            // make sure that the nick command is sent (expected to NOT run)
+            let sentNickCommand = false;
+            env.ircMock._whenClient(roomMapping.server, userIdNick, "send",
+                function (client, command, arg) {
+                    expect(client.nick).toEqual(userIdNick, "use the old nick on /nick");
+                    expect(client.addr).toEqual(roomMapping.server);
+                    expect(command).toEqual("NICK");
+                    expect(arg).toEqual(newNick);
+                    client._changeNick(userIdNick, newNick);
```
Comment (Half-Shot) [inline spec/integ/admin-rooms.spec.js:325]: You can remove these statements if you are just ensuring it doesn't run, fyi.

File: spec/integ/admin-rooms.spec.js [hunk 2]
```diff
@@ -308,6 +309,77 @@ describe("Admin rooms", function() {
         expect(sentSay).toBe(true, "sent say IRC command");
     }));
 
+    it("should be able to keep their name using !nick",
+        test.coroutine(function* () {
```
Comment (Half-Shot) [inline spec/integ/admin-rooms.spec.js:313]: We prefer async functions

```suggestion
        async () {
```

File: spec/integ/admin-rooms.spec.js (hunk 1 above)
Comment (Half-Shot) [reply spec/integ/admin-rooms.spec.js:325]: and below you can exchange the yields for await

File: spec/integ/admin-rooms.spec.js (hunk 2 above)
Comment (jaller94) [reply spec/integ/admin-rooms.spec.js:313]: `test.coroutine(function* ` stays consistent with all other tests in the file. May I find-replace it to async everywhere then?

Comment (jaller94) [reply spec/integ/admin-rooms.spec.js:313]: sounds like a separate short PR to me...

**Pull request merged**